tests/unit/cocoon/test_cocoon_complex_instrument.py

```diff
@@ -0,0 +1,204 @@
+                    "principal": "Principal",
+                    "coupon_rate": "Coupon",
+                },
+                {},
+                {"Isin": "ISIN", "ClientInternal": "Client Internal"},
+                [],
+                "TestPropertiesScope1",
+            ],
+        ]
+    )
+    def test_load_from_dataframe_bond(
+        self,
+        _,
+        scope,
+        file_name,
+        mapping_required,
+        mapping_optional,
+        identifier_mapping,
+        property_columns,
+        properties_scope,
+    ) -> None:
+        """
+            Test that instruments can be loaded successfully
+
+            :param str scope: The scope of the portfolios to load the transactions into
+            :param str file_name: The name of the test data file
+            :param dict{str, str} mapping_required: The dictionary mapping the dataframe fields to LUSID's required base transaction/holding schema
+            :param dict{str, str} mapping_optional: The dictionary mapping the dataframe fields to LUSID's optional base transaction/holding schema
+            :param dict{str, str} identifier_mapping: The dictionary mapping of LUSID instrument identifiers to identifiers in the dataframe
+            :param list[str] property_columns: The columns to create properties for
+            :param str properties_scope: The scope to add the properties to
+
+            :return: None
+            """
+        data_frame = pd.read_csv(Path(__file__).parent.joinpath(file_name))
+
+        responses = cocoon.cocoon.load_from_data_frame(
+            api_factory=self.api_factory,
+            scope=scope,
+            data_frame=data_frame,
+            mapping_required=mapping_required,
+            mapping_optional=mapping_optional,
+            file_type="instruments",
+            identifier_mapping=identifier_mapping,
+            property_columns=property_columns,
+            properties_scope=properties_scope,
+            instrument_name_enrichment=True,
+        )
+
+        expected_outcome = self.expected_response()
+
+        self.assertEqual(
+            first=sum(
+                [
+                    len(response.values)
+                    for response in responses["instruments"]["success"]
+                ]
+            ),
+            second=len(data_frame["Client Internal"].unique()),
+        )
+
+        self.assertEqual(
+            first=sum(
+                [
+                    len(response.failed)
+                    for response in responses["instruments"]["success"]
+                ]
+            ),
+            second=0,
+        )
+
+        combined_successes = {
+            correlation_id: instrument
+            for response in responses["instruments"]["success"]
+            for correlation_id, instrument in response.values.items()
+        }
+
+        self.assertTrue(
+            expr=all(
+                sorted(combined_successes[key].properties, key=lambda p: p.key)
```

Comment:
Does this test illustrate that your change is working, it looks to only check that all the property keys are the same between the test and the expected outcome?